diagnostics/TC_MSE/TC_snapshot_MSE_calc.py

```diff
@@ -0,0 +1,538 @@
+
+    hvar_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistvar_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempvar_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_LWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_OLRanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_SWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_RADanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_SEFanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_hflsanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hanom_hfssanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_LWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_OLRanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_SWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_RADanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_SEFanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_hflsanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hMoistanom_hfssanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_LWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_OLRanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_SWanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_RADanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_SEFanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_hflsanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+
+    hTempanom_hfssanom_save = np.ones((numstorms, numsteps, latlen, lonlen))*np.nan
+    
+    #Non-Radiative variables (ex: slp, wind, years, months, days, hours, latbox, lonbox, clat, clon, etc.)
+    #3D variables
+    latbox_save = np.ones((numstorms, numsteps, latlen))*np.nan
+
+    lonbox_save = np.ones((numstorms, numsteps, lonlen))*np.nan
+
+    #2D variables
+    maxwind_save = np.ones((numstorms, numsteps))*np.nan
+
+    minSLP_save = np.ones((numstorms, numsteps))*np.nan
+
+    Clat_save = np.ones((numstorms, numsteps))*np.nan
+
+    Clon_save = np.ones((numstorms, numsteps))*np.nan
+
+    year_save = np.ones((numstorms, numsteps))*np.nan
+
+    month_save = np.ones((numstorms, numsteps))*np.nan
+
+    day_save = np.ones((numstorms, numsteps))*np.nan
+
+    hour_save = np.ones((numstorms, numsteps))*np.nan
+
+    #Start Looping through the storms in the given year
+    for s, storm in enumerate(tracks_by_year[year]):
```

Comment:
This is fine as-is, but just pointing out that there are potentially efficiency gains in this block by leveraging more matrix-based operations and list comprehension in order to eliminate this for-loop.

Reply:
Thank you for giving this a check and the suggestions! I will certainly keep this in mind going forward. I changed the .rst file to add the CMOR names to what was there before.